### ISSUE TITLE
Pty dependency failing on Windows 10

### ISSUE THREAD

**Issue by RailsCod3rFuture** [mileszs/wicked_pdf#819]
## Issue description

When  I install the gem, I receive the error below:

I know that this error is the result of a LINUX based script, and I cannot run it on Windows 10.

```
Error:[rake --tasks] rake aborted!
LoadError: cannot load such file -- pty
C:/Ruby24-x64/lib/ruby/gems/2.4.0/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:283:in `require'
C:/Ruby24-x64/lib/ruby/gems/2.4.0/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:283:in `block in require'
C:/Ruby24-x64/lib/ruby/gems/2.4.0/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:249:in `load_dependency'
C:/Ruby24-x64/lib/ruby/gems/2.4.0/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:283:in `require'
C:/Ruby24-x64/lib/ruby/gems/2.4.0/gems/wicked_pdf-1.2.1/lib/wicked_pdf/progress.rb:3:in `<module:Progress>'
C:/Ruby24-x64/lib/ruby/gems/2.4.0/gems/wicked_pdf-1.2.1/lib/wicked_pdf/progress.rb:2:in `<class:WickedPdf>'
C:/Ruby24-x64/lib/ruby/gems/2.4.0/gems/wicked_pdf-1.2.1/lib/wicked_pdf/progress.rb:1:in `<top (required)>'
```
## Expected or desired behavior

When I install the gem, it should successfully complete its rake tasks.

## System specifications

```
wicked_pdf (1.2.1)
  wicked_pdf (~> 1.2, >= 1.2.1)
```

**https://rubygems.org**

platform/distribution and version (e.g. Windows 10/Heroku)

Is this dependency needed? Can it be removed.

**Comment by unixmonkey** [comment]
Yes, you should be able to opt out of using `pty` on windows systems.

I think we should add a guard around [this line](https://github.com/mileszs/wicked_pdf/blob/master/lib/wicked_pdf.rb#L30), and not require `progress` on windows, with a check similar to the one [a few lines above it](https://github.com/mileszs/wicked_pdf/blob/master/lib/wicked_pdf.rb#L8).

The Ruby 1.9 check shouldn't be necessary here, but `/mswin|mingw/` is the important part. Both are valid values on Windows systems, depending on how Ruby is installed.

I'd love it if you could try this out yourself, since I don't have easy access to a Windows machine right now, and send a pull request or patch.

**Comment by RailsCod3rFuture** [comment]
Thanks!

**Comment by unixmonkey** [comment]
I'm going to reopen this issue until either I see a related PR, or I get around to fixing it, since this is a newly-introduced issue from adding the progress feature.

**Comment by Lord-Fisch** [comment]
I created pull request #820 for this issue.
Verified working on Windows 10; Ruby 2.3.3; Rails 5.1.6.2

**Comment by unixmonkey** [comment]
Closing since this should be fixed in version `1.2.2` by the work in #820 . 

Thank you @Lord-Fisch !

If for some reason this isn't helping, please let me know and I'll reopen.
